src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1062,6 +1062,43 @@ private bool ImportIntrinsicCall(MethodDesc method)
 
             switch (method.Name)
             {
+                case "InitializeArray":
+                    if (metadataType.Namespace == "System.Runtime.CompilerServices" && metadataType.Name == "RuntimeHelpers")
+                    {
+                        var fieldSlot = (LdTokenEntry<FieldDesc>)_stack.Pop();
+                        var arraySlot = _stack.Pop();
+
+                        var node = (BlobNode)_compilation.GetFieldRvaData(fieldSlot.LdToken);
+
+                        LLVMValueRef src = LoadAddressOfSymbolNode(node);
+                        _dependencies.Add(node);
+                        int srcLength = node.GetData(_compilation.NodeFactory, false).Data.Length;
+
+                        LLVMValueRef arrayObjPtr = arraySlot.ValueAsType(LLVM.PointerType(LLVM.Int8Type(), 0), _builder);
+
+                        var argsType = new LLVMTypeRef[]
+                        {
+                            LLVM.PointerType(LLVM.Int8Type(), 0),
+                            LLVM.PointerType(LLVM.Int8Type(), 0),
+                            LLVM.Int32Type(),
+                            LLVM.Int32Type(),
+                            LLVM.Int1Type()
+                        };
+                        var memcpyFunction = GetOrCreateLLVMFunction("llvm.memcpy.p0i8.p0i8.i32", LLVM.FunctionType(LLVM.VoidType(), argsType, false));
+
+                        var args = new LLVMValueRef[]
+                        {
+                            LLVM.BuildGEP(_builder, arrayObjPtr, new LLVMValueRef[] { ArrayBaseSize() }, String.Empty),
+                            LLVM.BuildBitCast(_builder, src, LLVM.PointerType(LLVM.Int8Type(), 0), String.Empty),
+                            BuildConstInt32(srcLength),
+                            BuildConstInt32(0), // Assume no alignment
```

Comment:
@MichalStrehovsky, is any alignment guaranteed for the source? The destination is guaranteed to be pointer aligned.

Reply:
We pointer-align the buffer, but I'm not aware of a requirement to have these pointer-aligned (should we use the WASM backend as a JIT that directly reads the RVA data from the source IL, as opposed to reading the data from an AOT-compiled image where we control the alignment).

https://github.com/dotnet/corert/blob/5ec6a6b98da93620daec56de0100b0076c8a24d0/src/ILCompiler.Compiler/src/Compiler/Compilation.cs#L111-L112